32bit support

building on 32bit systems (respberry pi) is currently not possible.

```
GOARCH=arm go build
# github.com/bogem/id3v2
../../bogem/id3v2/util.go:43:10: constant 4261412864 overflows int
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,16 +1,18 @@
 language: go
 go:
-  - 1.5.4
-  - 1.6.4
   - 1.7.5
   - 1.8.1
+  - 1.9
 
-os:
-  - osx
-  - linux
+env:
+  matrix:
+    - GOOS=darwin
+    - GOOS=linux
+    - GOOS=windows
+    - GOARCH=arm GOARM=6
 
 script:
-  - go build && go test -v ./...
+  - go build && go test -race -v
 
 notifications:
   email: false

--- a/header.go
+++ b/header.go
@@ -61,7 +61,7 @@ func isID3Tag(data []byte) bool {
 	return bytes.Equal(data, id3Identifier)
 }
 
-func writeTagHeader(bw *bufio.Writer, framesSize int, version byte) error {
+func writeTagHeader(bw *bufio.Writer, framesSize uint, version byte) error {
 	// Identifier
 	if _, err := bw.Write(id3Identifier); err != nil {
 		return err

--- a/tag.go
+++ b/tag.go
@@ -364,7 +364,7 @@ func (tag *Tag) WriteTo(w io.Writer) (n int64, err error) {
 	// Write tag header.
 	bw := getBufioWriter(w)
 	defer putBufioWriter(bw)
-	if err := writeTagHeader(bw, framesSize, tag.version); err != nil {
+	if err := writeTagHeader(bw, uint(framesSize), tag.version); err != nil {
 		return 0, err
 	}
 	n += tagHeaderSize
@@ -383,7 +383,7 @@ func (tag *Tag) WriteTo(w io.Writer) (n int64, err error) {
 }
 
 func writeFrame(bw *bufio.Writer, id string, frame Framer) (int64, error) {
-	if err := writeFrameHeader(bw, id, frame.Size()); err != nil {
+	if err := writeFrameHeader(bw, id, uint(frame.Size())); err != nil {
 		return 0, err
 	}
 
@@ -391,7 +391,7 @@ func writeFrame(bw *bufio.Writer, id string, frame Framer) (int64, error) {
 	return frameHeaderSize + frameSize, err
 }
 
-func writeFrameHeader(bw *bufio.Writer, id string, frameSize int) error {
+func writeFrameHeader(bw *bufio.Writer, id string, frameSize uint) error {
 	// ID
 	if _, err := bw.WriteString(id); err != nil {
 		return err

--- a/util.go
+++ b/util.go
@@ -27,7 +27,7 @@ var (
 // writeBytesSize writes size to bw in form of ID3v2 size format (4 * 0b0xxxxxxx).
 //
 // If size is greater than allowed (256MB), then it returns ErrSizeOverflow.
-func writeBytesSize(bw *bufio.Writer, size int) error {
+func writeBytesSize(bw *bufio.Writer, size uint) error {
 	if size > maxSize {
 		return ErrSizeOverflow
 	}
@@ -40,7 +40,7 @@ func writeBytesSize(bw *bufio.Writer, size int) error {
 	// E.g. size is 32-bit integer and after the skip of first 4 bits
 	// its value is "10100111 01110101 01010010 11110000".
 	// In loop we should write every first 7 bits to bw.
-	mask := 254 << (3 * 8) // 11111110 000000000 000000000 000000000
+	mask := uint(254 << (3 * 8)) // 11111110 000000000 000000000 000000000
 	for i := 0; i < id3SizeLen; i++ {
 		// To take first 7 bits we should do `size&mask`.
 		firstBits := size & mask

--- a/util_test.go
+++ b/util_test.go
@@ -11,8 +11,8 @@ import (
 )
 
 var (
-	sizeInt   = 15351
-	sizeBytes = []byte{0, 0, 0x77, 0x77}
+	sizeInt   uint = 15351
+	sizeBytes      = []byte{0, 0, 0x77, 0x77}
 )
 
 func TestWriteBytesSize(t *testing.T) {

